redisinsight/ui/src/components/monitor-config/MonitorConfig.tsx

```diff
@@ -87,8 +87,12 @@ const MonitorConfig = ({ retryDelay = 10000 } : IProps) => {
     })
 
     // Catch connect error
-    newSocket.on(SocketEvent.ConnectionError, (error: Error) => {
-      payloads.push({ isError: true, time: `${Date.now()}`, message: `${error?.name}: ${error?.message}` })
+    newSocket.on(SocketEvent.ConnectionError, (error: { type: string; name: any; message: any }) => {
+      if (error?.type === 'TransportError') {
```

Comment:
Could you create a enum for constant string? May be in the future any type will added

Reply:
and better to use switch case insted of `if else` in the future

Reply:
> Could you create a enum for constant string? May be in the future any type will added

done

Reply:
> and better to use switch case instead of `else if` in the future

according to the eslint rules, we must have at least one `else if`, otherwise the `swith` construct is useless (as in current case)